packages/cloud_firestore/cloud_firestore_web/lib/transaction_web.dart

```diff
@@ -0,0 +1,49 @@
+part of cloud_firestore_web;
+
+class TransactionWeb implements Transaction {
+  final web.Transaction _webTransaction;
+  @override
+  FirestorePlatform firestore;
+
+  TransactionWeb._(this._webTransaction, this.firestore);
+
+  @override
+  Future<void> delete(DocumentReference documentReference) async {
+    assert(documentReference is DocumentReferenceWeb);
+    await _webTransaction.delete((documentReference as DocumentReferenceWeb).delegate);
+  }
+
+  @override
+  Future<DocumentSnapshot> get(DocumentReference documentReference) async {
+    assert(documentReference is DocumentReferenceWeb);
+    final webSnapshot = await _webTransaction.get((documentReference as DocumentReferenceWeb).delegate);
+    return _fromWeb(webSnapshot);
+  }
+
+  @override
+  Future<void> set(DocumentReference documentReference, Map<String, dynamic> data) async {
+    assert(documentReference is DocumentReferenceWeb);
+    await _webTransaction.set((documentReference as DocumentReferenceWeb).delegate, data);
+  }
+
+  @override
+  Future<void> update(DocumentReference documentReference, Map<String, dynamic> data) async {
+    assert(documentReference is DocumentReferenceWeb);
+    await _webTransaction.update((documentReference as DocumentReferenceWeb).delegate, data: data);
+  }
+
+  DocumentSnapshot _fromWeb(web.DocumentSnapshot webSnapshot) => DocumentSnapshot(
+      webSnapshot.ref.path,
+      webSnapshot.data(),
+      SnapshotMetadata(
+        webSnapshot.metadata.hasPendingWrites,
+        webSnapshot.metadata.fromCache,
+      ),
+      this.firestore);
+
+  @override
+  Future<void> finish() {
+    return Future.value();
+  }
```

Comment:
Hmmm, I don't understand this one very well... Are transactions in web finishing automatically, so this is just a noop implementation of the finish method?

Reply:
Yes, transactions on web don't have timeouts or manual finishing.

Reply:
Wow, ok!